src/data_quality_utils/polygon_matching/polygon_matching.py

```
@@ -0,0 +1,198 @@
+    osm_features_df = gpd.GeoDataFrame(geometry=[], crs=osm_query_crs)
+    osm_features_base = ox.features_from_polygon(polygon_for_osm, osm_tags)
+    # Check of standard formats.
+    osm_features_df = osm_features_base[
+        osm_features_base.geometry.geom_type.isin(
+            ["Polygon", "MultiPolygon", "LineString"]
+        )
+    ]
+    # This ensures that LineStrings are also interpreted as v.small polygons - shortcut for all in one solution.
+    osm_features_df = osm_features_df.to_crs(brdr_crs)
+    line_mask = osm_features_df["geometry"].geom_type == "LineString"
+    osm_features_df.loc[line_mask, "geometry"] = osm_features_df.loc[
+        line_mask, "geometry"
+    ].buffer(line_buffer)
+
+    # Buffer of zero is to ensure it is a GeoPandas series, rather than a df with other args.
+    osm_features_df = osm_features_df.buffer(0)
+
+    # Align OSM co-ordinate systems
+    print(f"Downloaded {len(osm_features_df)} polygons.")
+
+    # Final check for crs matching - maybe not necessary
+    # FUTURE OLI LOOK AT THIS
+    if osm_features_df.crs == brdr_crs:
+        osm_features_proj = osm_features_df
+    else:
+        osm_features_proj = osm_features_df.to_crs(brdr_crs)
+    return osm_features_proj
+
+
+def process_osm_polygons(osm_features_proj, used_osm_indices):
+    reference_geom = None
+    geometries_to_combine = None
+
+    if used_osm_indices is None:
+        # Here we use ALL downloaded features
+        print("Using all polygons.")
+        geometries_to_combine = osm_features_proj.geometry
+    elif isinstance(used_osm_indices, list) and used_osm_indices:
+        # Here we only use features at specific indices
+        print(f"Selecting polygons at indices {used_osm_indices}.")
+        geometries_to_combine = osm_features_proj.iloc[used_osm_indices].geometry
+
+    # Combine all the geometries we need
+    combined_reference = unary_union(geometries_to_combine)
+    if not combined_reference.is_empty:
```

Is this a problem? Should we raise an error or a warning?